### PR TITLE
Completed Item model and Item GET express routes

### PR DESCRIPTION
**Pull request by 74tevans**

### CODE REVIEW

File: server/models/Item.js [hunk 1]
```diff
@@ -5,7 +5,32 @@ class Item extends Model {}
...
 Item.init(
   {
-    // Define your columns here
+    name: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      unique: true // Users will search for items by name, so name must be unique
+    },
+    price: {
+      type: DataTypes.NUMBER,
```
Comment (sanya-pramod) [inline server/models/Item.js:14]: DataTypes.NUMBER not a valid sequelize datatype, use DataTypes.FLOAT

File: server/models/Item.js [hunk 2]
```diff
@@ -5,7 +5,32 @@ class Item extends Model {}
...
+      allowNull: false
+    },
+    category: {
+      type: DataTypes.STRING,
+      allowNull: false
+    },
+    image: {
+      type: DataTypes.STRING,
+      validate: {
+        isUrl: true // All images in seed data are URLs, so validate image entry is a URL
```
Comment (sanya-pramod) [inline server/models/Item.js:31]: Perhaps add an allowNull here as we are adding new items to the inventory, the "user" may not have a selsected image yet (can always add through update)

File: server/routes/items.js [hunk 3]
```diff
@@ -3,7 +3,34 @@ const { Item } = require("../models");
...
+router.get('/:id', async (req, res) => {
+    try {
+        const item = await Item.findByPk(req.params.id);
+        if (!item) {
+            res.status(404).json({error: "Item not found"});
+            return;
+        }
+        res.status(200).json(item);
+    } catch (error){
+        next(error);
```
Comment (sanya-pramod) [inline server/routes/items.js:32]: in both router.get handlers, you are using next(error), but next is not included as a parameter in your route handlers - runtime error:
router.get('/', async (req, res, next) => { ... });
router.get('/:id', async (req, res, next) => { ... });

File: server/routes/items.js [hunk 4]
```diff
@@ -3,7 +3,34 @@ const { Item } = require("../models");
...
+    }
+});
+
+// Get the specified item from parameters and return OK status. If any errors occur or item doesn't exist, throw the error.
+router.get('/:id', async (req, res) => {
+    try {
+        const item = await Item.findByPk(req.params.id);
+        if (!item) {
+            res.status(404).json({error: "Item not found"});
+            return;
```
Comment (sanya-pramod) [inline server/routes/items.js:28]: Item.findAll() will always return an array (even if empty), so if (!items) will never be true. - potential error
do: 
if (items.length === 0){
...
}

**Review by sanya-pramod — CHANGES_REQUESTED** [review]
Just some small stuff needs to be changed :)

**Review by 74tevans — COMMENTED** [review]
All issues fixed

**Pull request merged**